libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/IncrementalCompilationMultiProjectIT.kt

```diff
@@ -687,62 +649,78 @@ abstract class BaseIncrementalCompilationMultiProjectIT : IncrementalCompilation
         }
     }
 
-    @DisplayName("KT-49780: Valid outputs after cache corruption exception")
+    @DisplayName("Test handling of failure caused by user error")
     @GradleTest
-    open fun testValidOutputsWithCacheCorrupted(gradleVersion: GradleVersion) {
+    fun testFailureHandling_UserError(gradleVersion: GradleVersion) {
         defaultProject(gradleVersion) {
-            breakCachesAfterCompileKotlinExecution(this)
-
-            build("assemble")
+            build(":app:compileKotlin")
 
-            subProject("lib").kotlinSourcesDir().resolve("bar/B.kt").modify {
-                it.replace("fun b() {}", "fun b() = 123")
+            // Make a compile error in the source code
+            var originalContents: String? = null
+            subProject("lib").kotlinSourcesDir().resolve("bar/A.kt").modify {
+                originalContents = it
+                it.replace("fun a() {}", "fun a() { // Compile error: Missing closing bracket")
             }
 
-            build("assemble") {
-                assertOutputContains("Non-incremental compilation will be performed: INCREMENTAL_COMPILATION_FAILED")
+            // In the next build, incremental compilation should fail
+            buildAndFail(":app:compileKotlin") {
+                assertTasksFailed(":lib:$compileKotlinTaskName")
+                assertOutputContains("Compilation error. See log for more details")
+                // Also check that we did not fall back to non-incremental compilation
+                assertOutputDoesNotContain("Non-incremental compilation will be performed: ${BuildAttribute.INCREMENTAL_COMPILATION_FAILED.name}")
             }
 
-            val lookupFile = projectPath.resolve("lib/build/kotlin/${compileKotlinTaskName}/cacheable/${compileCacheFolderName}/lookups/file-to-id.tab")
-            assertTrue("Output is empty", lookupFile.exists())
+            // Fix the compile error in the source code
+            subProject("lib").kotlinSourcesDir().resolve("bar/A.kt").writeText(originalContents!!)
+            changeMethodSignatureInLib()
 
+            // In the next build, compilation should be incremental and succeed
+            build(":app:compileKotlin") {
+                assertIncrementalCompilation(expectedCompiledKotlinFiles =
+                    getExpectedKotlinSourcesForDefaultProject(
+                        libSources = listOf("bar/A.kt", "bar/B.kt", "bar/barUseA.kt"),
+                        appSources = listOf("foo/AA.kt", "foo/AAA.kt", "foo/BB.kt", "foo/fooUseA.kt")
+                    )
+                )
+            }
         }
     }
 
-    @DisplayName("KT-49780: No need to rebuild invalid code due to cache corruption")
+    @DisplayName("Test handling of failure caused by tool error")
     @GradleTest
-    open fun testIncrementalBuildWithCompilationError(gradleVersion: GradleVersion) {
```

Comment:
@nav-nav I've removed this test because it should already be covered by the existing `testCompileErrorInLib` (now renamed to `testFailureHandling_UserError`). If you see any issues with this PR, please let me know.